### PR TITLE
Modify the grove rpm version format

### LINKED ISSUE #2468: Update Grove version to match TC
Currently the version number for Grove does not follow the conventions of the rest of traffic control.

### PR DESCRIPTION
**Pull request by jrushford**
Modify the grove/build/build_rpm.sh to change the version format to add the last 9 digit commit hash on the grove/ directory as the releasewhen building the grove rpm.  Fixes #2468

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1932/
Test PASSed.

**Comment by dangogh** [comment]
Can you modify the title to describe the change?  something like 'change grove version format' would be better when looking at a long list of PRs.   And I assume this is to fix the issue @jhg03a filed on the grove version?  Put `Fixed #xxxx` in the description with that issue number so it'll be closed automatically when this is merged.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1934/
Test PASSed.

File: grove/build/build_rpm.sh [hunk 1]
```diff
@@ -19,6 +19,7 @@ cd "$ROOTDIR/grove"
 
 BUILDDIR="$ROOTDIR/grove/rpmbuild"
 VERSION=`cat ./VERSION`.`git rev-list --all --count`
+RELEASE=`git log -1 | awk '/^commit/ {print substr($2,1,9)}'`
```
Comment (dangogh) [inline grove/build/build_rpm.sh]: This doesn't get the same form as the other components, so the release id won't match.    In `trafficcontrol/build/README.md` describes how this is done for the other components -- there are functions in `trafficcontrol/build/functions.sh` that do this for you -- I recommend following the recipe in the other components to get the same result.

I do understand not wanting to have the grove main version number be different from the others (e.g. 0.1.x vs 2.3.x),  but the commit count and commit hash should be the same.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1977/
Test FAILed.

**Comment by jrushford** [comment]
@dangogh  I re-worked the build_rpm.sh to use the functions.sh and build the rpm like the others are built.  The grove version number is different however and that should not be a problem.  the build number and git hash follows the same convention as other tc rpmbuilds.

**Comment by jrushford** [comment]
[approve ci]

**Comment by dangogh** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/1983/
Test PASSed.

**Pull request merged**